src/main/java/org/opensearch/security/securityconf/ConfigModelV7.java

```diff
@@ -1116,19 +1116,19 @@ public Map<String, Boolean> mapTenants(final User user, Set<String> roles) {
 
                 if (rw || !result.containsKey(tenant)) { //RW outperforms RO
 
-                    // We want to make sure that we add a tenant that exissts
+                    // We want to make sure that we add a tenant that exists
                     // Indeed, because we don't have control over what will be
                     // passed on as values of users' attributes, we have to make
                     // sure that we don't allow them to select tenants that do not exist.
-                    if(ConfigModelV7.this.tenants.getCEntries().keySet().contains(tenant)) {
+                    if(ConfigModelV7.this.tenants.getCEntries().containsKey(tenant)) {
                         result.put(tenant, rw);
                     }
                 }
             });
             
             if(!result.containsKey("global_tenant") && (
                     roles.contains("kibana_user")
-                    || roles.contains("kibana_user")
+                    || roles.contains("KIBANA_USER")
```

Comment:
How would you feel about making this check even better by changing the collection to be a `new TreeMap<>(String.CASE_INSENSITIVE_ORDER)`, then the two checks per case can be changed into one?

Reply:
make sense, i have follow the suggestion and made the change.